Kronos/FileSystem/FileSystem.php

```diff
@@ -16,14 +17,21 @@ class FileSystem implements FileSystemInterface {
 	 * @var Selector
 	 */
 	private $mountSelector;
+
 	/**
 	 * @var FileRepositoryInterface
 	 */
 	private $fileRepository;
 
-	public function __construct(Selector $mountSelector, FileRepositoryInterface $fileRepository) {
+	/**
+	 * @var MetadataTranslator
+	 */
+	private $metadataTranslator;
+
+	public function __construct(Selector $mountSelector, FileRepositoryInterface $fileRepository,MetadataTranslator $metadataTranslator = null) {
 		$this->mountSelector = $mountSelector;
 		$this->fileRepository = $fileRepository;
+		$this->metadataTranslator = is_null($metadataTranslator) ? new MetadataTranslator() : $metadataTranslator;
```

Comment:
`$this->metadataTranslator = ($metadataTranslator ?: new MetadataTranslator());`